mongo/integration/client_side_encryption_prose_test.go

```diff
@@ -737,8 +806,14 @@ func TestClientSideEncryptionProse(t *testing.T) {
 				invalidKeyOpts := options.DataKey().SetMasterKey(tc.masterKey)
 				_, err = invalidClientEncryption.CreateDataKey(mtest.Background, tc.provider, invalidKeyOpts)
 				assert.NotNil(mt, err, "expected CreateDataKey error, got nil")
-				assert.True(mt, strings.Contains(err.Error(), "parse error"),
-					"expected error %v to contain substring 'parse error'", err)
+				if tc.provider == "kmip" {
```

Comment:
Isn't the error substring already checked on [L781](https://github.com/mongodb/mongo-go-driver/pull/812/files#diff-55617555c17bae6fadeb3e9a0c0e3a10837496998c75937eba67712b871f58eeL712)? Do we even need to call `CreateDataKey` again here and check the error?

Reply:
I think that this `CreateDataKey()` is called on a client encryption object with invalid options instead but might be misunderstanding this test.

Reply:
> Do we even need to call CreateDataKey again here and check the error?

Yes. Some test cases call `CreateDataKey` on the `invalidClientEncryption` object. See https://github.com/mongodb/specifications/tree/master/source/client-side-encryption/tests#test-cases.

The tests previously assumed the call to `CreateDataKey` with `testInvalidClientEncryption` always returned an error with the substring "parse error". That is no longer true for the "kmip" case.

I think this is correct. I suggest an optional improvement to add a field to the `testCases` for the expected error substring on `testInvalidClientEncryption`:

```go
testCases := []struct {
			name                                  string
			provider                              string
			masterKey                             interface{}
			errorSubstring                        string
			testInvalidClientEncryption           bool
			invalidClientEncryptionErrorSubstring string
		}
```

But I am OK with this as-is.

Reply:
This looks cleaner! Fixed.